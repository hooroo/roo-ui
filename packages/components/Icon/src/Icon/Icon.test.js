import React from 'react';
import { shallow } from 'enzyme';

import { Icon } from '.';

describe('<Icon />', () => {
  let wrapper;
  const iconPath = 'M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z';

  it('renders correctly', () => {
    expect(shallow(<Icon path={iconPath} />)).toMatchSnapshot();
  });

  it('renders the <path /> for the provided icon', () => {
    wrapper = shallow(<Icon path={iconPath} />);
    expect(wrapper.find('path').props()).toHaveProperty('d', iconPath);
  });

  describe('with the size prop', () => {
    beforeEach(() => {
      wrapper = shallow(<Icon path={iconPath} size="3rem" />);
    });

    it('sets the width and height attributes of the <svg /> with the size', () => {
      expect(wrapper.find('svg').props()).toMatchObject({
        height: '3rem',
        width: '3rem',
      });
    });
  });

  describe('with the width prop', () => {
    beforeEach(() => {
      wrapper = shallow(<Icon path={iconPath} width="3rem" />);
    });

    it('sets the width attribute of the <svg />', () => {
      expect(wrapper.find('svg').props()).toMatchObject({
        height: '1em',
        width: '3rem',
      });
    });
  });

  describe('with the height prop', () => {
    beforeEach(() => {
      wrapper = shallow(<Icon path={iconPath} height="3rem" />);
    });

    it('sets the height attribute of the <svg />', () => {
      expect(wrapper.find('svg').props()).toMatchObject({
        height: '3rem',
        width: '1em',
      });
    });
  });

  describe('with the title prop', () => {
    beforeEach(() => {
      wrapper = shallow(<Icon path={iconPath} title="An accessible title" />);
    });

    it('renders a <title /> with the title text', () => {
      expect(wrapper.find('title').text()).toBe('An accessible title');
    });
  });

  describe('with the fill prop', () => {
    beforeEach(() => {
      wrapper = shallow(<Icon path={iconPath} fill="#c00" />);
    });

    it('sets the fill attribute of the <svg />', () => {
      expect(wrapper.find('svg').props()).toHaveProperty('fill', '#c00');
    });
  });
});
