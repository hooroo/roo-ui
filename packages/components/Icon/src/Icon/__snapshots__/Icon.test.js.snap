// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    fill="currentColor"
    height={null}
    path="M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z"
    size="1em"
    title={null}
    width={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <path
          d="M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z"
        />,
      ],
      "fill": "currentColor",
      "height": "1em",
      "viewBox": "0 0 36 36",
      "width": "1em",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "d": "M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z",
        },
        "ref": null,
        "rendered": null,
        "type": "path",
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <path
            d="M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z"
          />,
        ],
        "fill": "currentColor",
        "height": "1em",
        "viewBox": "0 0 36 36",
        "width": "1em",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M12.272 25.036l-8.749-8.49L0 20.026 12.272 32 36 8.482 32.474 5z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
